docs/source/formulations_example.rst

```diff
@@ -0,0 +1,648 @@
+The predictor takes care of flattening it.
+
+.. code-block:: python
+
+    from citrine.informatics.design_spaces import FormulationDesignSpace
+    from citrine.informatics.constraints import IngredientCountConstraint, IngredientFractionConstraint
+
+    fds = FormulationDesignSpace(
+        name="margaritas formulation,
+        description="",
+        formulation_descriptor=formulation,
+        ingredients={"simple syrup A", "simple syrup B", "tequila", "ice", "triple sec", "lime juice"},
+        labels={
+            "acid": {"lime juice"},
+            "alcohol": {"tequila", "triple sec"},
+            "sweetener": {"triple sec"},
+            "simple syrup": {"simple syrup A", "simple syrup B"}
+        },
+        constraints={
+            IngredientCountConstraint(formulation_descriptor=formulation, min=3, max=5),
+            IngredientCountConstraint(formulation_descriptor=formulation, min=1, max=1, label="simple syrup"),
+            IngredientFractionConstraint(formulation_descriptor=formulation, ingredient="lime juice", min=0.15, max=0.30)
+        }
+    )
+
+But that's not all; we also have to define the ``blend time`` input.
+We do this by wrapping a :class:`~citrine.informatics.design_spaces.product_design_space.ProductDesignSpace` around the formulation design space and a ``blend time`` dimension.
+
+.. code-block:: python
+
+    from citrine.informatics.design_spaces import ProductDesignSpace
+
+    design_space = ProductDesignSpace(
+        name="margaritas design space",
+        description="",
+        subspaces=[fds],
+        dimensions=[ContinuousDimension(descriptor=blend_time, lower_bound=5.0, upper_bound=20.0)]
+    )
+    design_space = project.design_spaces.register(design_space)
+
+Proposing New Formulation Candidates
+------------------------------------
+
+With the pieces assembled, we define a :class:`~citrine.informatics.workflows.design_workflow.DesignWorkflow`.
+Our goal is to find the margarita that is most likely to have a tastiness score above 7.5, while keeping the cost per kg below $12.50.
+We define an :class:`~citrine.informatics.scores.LIScore` with this objective and constraint, define a design workflow with the predictor and design space, and trigger the design workflow on the score.
+
+.. code-block:: python
+
+    from citrine.informatics.scores import LIScore
+    from citrine.informatics.objectives import ScalarMaxObjective
+    from citrine.informatics.constraints import ScalarRangeConstraint
+    from citrine.informatics.workflows.design_workflow import DesignWorkflow
+
+    score = LIScore(
+        objectives=[ScalarMaxObjective(descriptor_key=blend_time.key)],
+        baselines=[7.5],
+        constraints=[ScalarRangeConstraint(
+            descriptor_key=price_descriptors[0].key, upper_bound=12.50, lower_bound=0.00
+        )]
+    )
+
+    design_workflow = DesignWorkflow(
+        name="best margarita below 12.50",
```

Comment:
SHOULD - we should probably not include targets in the name of the design workflow, since they are independent and may confuse people into making a new design workflow every time they want to change targets.